pages/_en-US/ds-index/ds-bios-firmware-dump.md

```diff
@@ -0,0 +1,75 @@
+---
+lang: en-US
+layout: wiki
+section: ds-index
+category: guides
+title: BIOS / Firmware Dumping
+description: How to extract the Nintendo DS or DSi BIOS / Firmware from your console
+tabs:
+  - tab-twl-sd-card: DSi / 3DS with custom firmware
+    tab-flashcard: Flashcard
+---
+
+Some emulators, such as MelonDS, require the DS-mode or DSi-mode BIOS and firmware, which is only available by extracting from your console. There are two distinct ways to achieve this, being split into a DSi or 3DS with custom firmware, or a DS / DS Lite / DSi or 3DS without custom firmware and a flashcard compatible with the console.
+
+The DSi / 3DS method will extract both the DS-mode and DSi-mode BIOS and firmware files. The Flashcard method will only extract the DS-mode files. It is highly recommended to use the DSi / 3DS method where possible.
+
+{% capture tab-twl-sd-card %}
+
+This page assumes you are running a modern CFW environment from [3ds.hacks.guide](https://3ds.hacks.guide) or [dsi.cfw.guide](https://dsi.cfw.guide).
+{:.alert .alert-warning}
+
+This guide assumes you have [TWiLight Menu++](../twilightmenu/installing) installed on your device.
+{:.alert .alert-warning}
+
+### Part 1: Getting the required files
+
+1. Download [dsibiosdumper](https://melonds.kuribo64.net/downloads/dsibiosdumper.7z)
+1. Copy `dsibiosdumper.nds` from the dsibiosdumper `.7z` file to the root of your SD card
+1. Reinsert your SD card into your device
+
+### Part 2: dsibiosdumper
+1. Power on your device
+1. Launch TWiLight Menu++
```

Comment:
dsibiosdumper should always be run from Unlaunch if at all possible, dumps from TWiLight Menu++ will be incomplete same as dumps made on 3DS

Reply:
How do you suggest we run this on a 3DS? It should work there, but we don't really have a valid way to run it other than some sorta homebrew launcher.

Then again, maybe it doesn't support dumping NAND, and thus would need a separate guide...

~~another tool maybe?~~

Reply:
TWiLight, but either 3DS should just be left out as a bad idea or separate from DSi

Reply:
does it work as a cia? lol

Reply:
it wouldn’t help